Pal/src/host/Linux-SGX/enclave_framework.c

```diff
@@ -847,56 +847,22 @@ void test_dh (void)
 }
 #endif
 
-#define RSA_KEY_SIZE        2048
-#define RSA_E               3
-
 int init_enclave (void)
 {
-    int ret;
-    LIB_RSA_KEY *rsa = malloc(sizeof(LIB_RSA_KEY));
-    lib_RSAInitKey(rsa);
-
-    ret = lib_RSAGenerateKey(rsa, RSA_KEY_SIZE, RSA_E);
-    if (ret != 0) {
-        SGX_DBG(DBG_S, "lib_RSAGenerateKey failed: %d\n", ret);
-        return ret;
-    }
-
-    PAL_NUM nsz = RSA_KEY_SIZE / 8, esz = 1;
-    uint8_t n[nsz], e[esz];
-
-    ret = lib_RSAExportPublicKey(rsa, e, &esz, n, &nsz);
-    if (ret != 0) {
-        SGX_DBG(DBG_S, "lib_RSAExtractPublicKey failed: %d\n", ret);
-        goto out_free;
-    }
-
-    LIB_SHA256_CONTEXT sha256;
-
-    ret = lib_SHA256Init(&sha256);
-    if (ret < 0)
-        goto out_free;
-
-    ret = lib_SHA256Update(&sha256, n, nsz);
-    if (ret < 0)
-        goto out_free;
-
-    ret = lib_SHA256Final(&sha256, (uint8_t *) pal_enclave_state.enclave_keyhash);
-    if (ret < 0)
-        goto out_free;
-
-    pal_enclave_config.enclave_key = rsa;
+    /*
+     * The enclave identifier is uniquely created for each enclave to
+     * prevent man-in-the-middle attack during local attestion.
+     * The identifier is injected into the local attestation report
+     * to be authenticated by another trustd process.
+     */
```

Comment:
What happens if you get the same value for two enclaves?  Can we detect this case, or pass in a pid from the parent that we have reason to believe is unique?

Reply:
The enclave id is a 256-bit random number, generated by the CPU's rng. The probability of collision on this id should be low enough given that enclave creation is quite slow. Relying on pids passed from the host wouldn't be secure in this scenario.

Reply:
Can you write a bit longer description of the protocol in a comment (or point me to it if it is somewhere else)?  I am not confident I understand this yet, and I think this would be good documentation.  It would help to know what steps are done in which enclave, what messages are exchanged (and if they are in plaintext), and how the decision is made to accept the connection.